### PR TITLE
Log: Improving the logs

### PR DESCRIPTION
**Pull request by eloycoto**
This PR use a new logging system, because it was hard to know when it
was an error level, info or debug, so make things supper hard to debug.

With the new tool, the following output will be created:
```
level=info msg="Data directory: /tmp/device/device"
level=info msg="device config file: /tmp/device/device/device-config.json"
level=error msg="open /tmp/device/device/device-config.json: no such file or directory"
level=debug msg="added target 'system' with the following scraper: '[HTTP scraper for URL: ]'"
level=debug msg="added target 'data transfer' with the following scraper: '[Prometheus object scraper]'"
level=info msg="started targetMetric for workload 'system'"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/_ping"
level=info msg="started targetMetric for workload 'data transfer'"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/info"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/pods/json"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/events"
level=debug msg="Update log transports, number of tranports: 0"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/pods/json"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/secrets/json"
level=debug msg="DoRequest Method: GET URI: http://d/v4.2.0/libpod/pods/json"
level=error msg="failed to run 'rpm-ostree status'"
level=warning msg="OS management is not available. OS configuration updates will have no effect."
level=debug msg="No OS management configuration. Not updating."
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=warning msg="Could not read wlp61s0 speed" error="read /sys/class/net/wlp61s0/speed: invalid argument"
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=debug msg="Reading file /sys/class/net/enp11s0u1/device/device" error="open /sys/class/net/enp11s0u1/device/device: no such file or directory"
level=debug msg="Reading file /sys/class/net/enp11s0u1/device/vendor" error="open /sys/class/net/enp11s0u1/device/vendor: no such file or directory"
level=error msg="cannot register device. DeviceID: eloy-test; err: rpc error: code = DeadlineExceeded desc = context deadline exceeded"
level=info msg="configuration has not been initialized yet. Sending registration request. DeviceID: eloy-test;"
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=warning msg="Could not read wlp61s0 speed" error="read /sys/class/net/wlp61s0/speed: invalid argument"
level=debug msg="biosdevname error: exec: \"biosdevname\": executable file not found in $PATH"
level=debug msg="Reading file /sys/class/net/enp11s0u1/device/device" error="open /sys/class/net/enp11s0u1/device/device: no such file or directory"
level=debug msg="Reading file /sys/class/net/enp11s0u1/device/vendor" error="open /sys/class/net/enp11s0u1/device/vendor: no such file or directory"
level=error msg="cannot register device. DeviceID: eloy-test; err: rpc error: code = DeadlineExceeded desc = context deadline exceeded"
```

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

File: internal/heartbeat/heartbeat.go [hunk 1]
```diff
@@ -5,14 +5,13 @@ import (
...
 	"net/http"
-	"os"
 	"sync"
 	"sync/atomic"
 	"time"
 
-	"git.sr.ht/~spc/go-log"
 	"github.com/go-openapi/strfmt"
 	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
```
Comment (gciavarrini) [inline internal/heartbeat/heartbeat.go:14]: Question: do we need to set the formatter once again?

File: internal/registration/gomock_reflect_1904518823/prog.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (gciavarrini) [inline internal/registration/gomock_reflect_1904518823/prog.go:1]: Should this file be part of the PR?

File: internal/heartbeat/heartbeat.go (hunk 1 above)
Comment (eloycoto) [reply internal/heartbeat/heartbeat.go:14]: No, no need to do that at all.

**Review by gciavarrini — COMMENTED** [review]
Is there any drawback to use `logrus` instead of `git.sr.ht/~spc/go-log`? Why was it selected?

**Review by pkliczewski — APPROVED** [review]
Looks good to me should we fix this linting issue in this PR or next one?

> SA1019: "io/ioutil" has been deprecated since Go 1.16

**Comment by gciavarrini** [comment]
IMO, it would be better to fix lint issue in this PR, but I've no a strong opinion :)

**Comment by eloycoto** [comment]
> Is there any drawback to use logrus instead of git.sr.ht/~spc/go-log? Why was it selected?

The main reason is that go-log does not support level printing and adding it is more work.

**Pull request merged**